### PR TITLE
Slight fix on `entropy` test

### PR DESCRIPTION
**Pull request by takagi**
This PR follows #4369 to slightly fix its test.

### CODE REVIEW

File: tests/cupyx_tests/scipy_tests/stats_tests/test_distributions.py [hunk 1]
```diff
@@ -124,11 +124,12 @@ def _entropy(self, xp, scp, dtype, shape, use_qk, base, axis, normalize):
...
 
     @testing.for_all_dtypes(no_complex=True)
-    @testing.numpy_cupy_allclose(rtol={cupy.float16: 1e-3,
-                                       cupy.float32: 1e-6,
+    @testing.numpy_cupy_allclose(rtol={cupy.float32: 1e-6,
                                        'default': 1e-15},
                                  scipy_name='scp')
     def test_entropy(self, xp, scp, dtype):
+        # `entropy` returns float32 value for float16 input so float32
+        # tolerance is used for it.
```
Comment (emcastillo) [inline tests/cupyx_tests/scipy_tests/stats_tests/test_distributions.py:132]: Are the intermediate computations done in float32?
If that is the case this is fine. But if all the steps are done in float16, even though the output is float32, there will be precision losses. So the original tolerance should be kept.

**Comment by takagi** [comment]
Yes, float16 input is `astype`ed to float32 at the very early point of
`entropy` (at least on CuPy.)

https://github.com/cupy/cupy/blob/3ce7aafc824863b35021862316d2b1b9b1a90736/cupyx/scipy/stats/distributions.py#L46

But the problem is that per-dtype tolerance uses the dtype of the returned
ndarray, so it can not determine that the input was float16 or floar32 when
the function returns float32 value for both float16 and float32 inputs...


2020年12月9日(水) 12:32 emcastillo <notifications@github.com>:

**Comment by takagi** [comment]
Maybe possible to make `numpy_cupy_allclose` to sniff `dtype` when it is
used with `for_dtypes` family at the same time?

**Comment by takagi** [comment]
Shuffle test failed as `axis` parameter is supported since scipy==1.4.0.

File: tests/cupyx_tests/scipy_tests/stats_tests/test_distributions.py (hunk 1 above)
Comment (grlee77) [reply tests/cupyx_tests/scipy_tests/stats_tests/test_distributions.py:132]: In SciPy, float16 inputs do not seem to be explicitly cast to float32, so the normalization step that divides by a sum over the array will use float16. Thus, the normalization step here should be more accurate than in SciPy. The later use of the `scipy.special` functions does result in things becoming float32, though (I don't think any functions in `scipy.special` support half precision)

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Seems like it didn't go well 😁

**Comment by toslunar** [comment]
Could you merge #4481 first?

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request closed without merging**